app/src/main/java/com/eleks/mowid/ui/feature/main/MainActivity.kt

```diff
@@ -1,23 +1,38 @@
 package com.eleks.mowid.ui.feature.main
 
 import android.os.Bundle
+import android.widget.Toast
 import androidx.activity.compose.setContent
 import androidx.activity.viewModels
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Surface
 import androidx.compose.ui.Modifier
+import androidx.core.app.ComponentActivity
+import com.eleks.mowid.R
 import com.eleks.mowid.base.ui.BaseActivity
 import com.eleks.mowid.ui.navigation.AppNavigation
 import com.eleks.mowid.ui.theme.MoWidTheme
+import com.firebase.ui.auth.AuthUI
+import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract
+import com.firebase.ui.auth.data.model.FirebaseAuthUIAuthenticationResult
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class MainActivity : BaseActivity<MainState, MainEvent, MainEffect, MainViewModel>() {
     override val viewModel: MainViewModel by viewModels()
 
     override fun handleEffect(effect: MainEffect) {
-        //TODO:
+        when (effect) {
+            MainEffect.SignIn -> createSignInIntent()
```

Comment:
As I see it, there is a minor misusing of Effects and its original intention. 
Effects are used to notify about one-time things that should happen on UI—for example, ShowToast. 
In this case, I suggest also reading an article about it - https://proandroiddev.com/mvi-architecture-with-kotlin-flows-and-channels-d36820b2028d
Also, I suggest creating a separate layer (UserRepository) to make Firebase-related logic. I know it is related to an Activity and its result, but it would still be better. 
What do you think?